### DISCUSSION TITLE
How to perform implementation of shared viewModel using precompose-koin?

### DISCUSSION THREAD

**Discussion by farhazulmullick-pw** [Tlaster/PreCompose#225] [Q&A]
Is there any way so that we can use same viewModel reference in all the screen of a group using precompose-koin instead of implementeing like.

`val vm by inject<MYViewModel>()` in every screen

**Comment by Tlaster** [comment]
Sorry but I don't get it, if you want to get something that is injected from koin, at least you need to call something like`val sth = koinInject<Something>()`, so in your case, you still need to call `val vm = koinViewModel<ViewModel>()` in every screen even you do can get the same instance of view model by passing the root `LocalStateHolder.current` to every screen like this `val vm = koinViewModel<ViewModel>(stateHolder = stateHolder)`

**Comment by farhazulMullick** [comment]
@Tlaster 
Is there any way to get the lifecycle of a graph/nested-graphs in a NavHost. Like when we are releasing the resources?
Is it kept in memory till NavHost composable's disposableEffect is called?

**Reply by Tlaster** [reply]
There's a much more simple way to do it, just use `DisposableEffect` at the same level of your `NavHost` like this:
```Kotlin
DisposableEffect(Unit) {
    //...
    onDispose {
        // release your resource
    }
}
NavHost(
    //...
)
```
If you want a per back stack's lifecycle, there's also a `LocalLifecycleOwner.current` that you can use.

**Reply by farhazulMullick** [reply]
Okay , thanks.
I wanted to scope my viewModel to a particular group in a NavHost? How can i do that

**Reply by Tlaster** [reply]
As I said before, you can pass the root `LocalStateHolder.current` to every screen, here is a complete example:
```Kotlin
val stateHolder = LocalStateHolder.current
NavHost(
    navigator = navigator,
    initialRoute = "/home",
) {
    scene("/home") {
        val viewModel = koinViewModel(NoteListViewModel::class, stateHolder = stateHolder)
    }
}
```
